src/main/java/org/sonarsource/sonarlint/ls/AnalysisTaskExecutor.java

```diff
@@ -383,7 +398,7 @@ private IssueListener createIssueListener(Map<URI, VersionedOpenFile> filesToAna
           var versionedOpenFile = filesToAnalyze.get(uri);
           if (issue.getType() == RuleType.SECURITY_HOTSPOT) {
             securityHotspotsCache.reportIssue(versionedOpenFile, issue);
-          } else {
+          } else if (!task.shouldKeepHotspotsOnly()) {
```

Comment:
Maybe just pass boolean here? Not the full task object, since we don't really need the full object.